### PR TITLE
feat(smdk): add smdk clean

### LINKED ISSUE #3985: [feature] add `smdk clean`
Mapped to `cargo clean`

### PR DESCRIPTION
**Pull request by diogopontual**
I'm trying to learn the fluvio code.

I got the first "good first issue" on the list.

Fixes #3985

### CODE REVIEW

**Comment by diogopontual** [comment]
Hi @EstebanBorai. I just noticed there is a problem with the PR. Do you have any clue how I can fix it? TY.

**Comment by EstebanBorai** [comment]
> Hi @EstebanBorai. I just noticed there is a problem with the PR. Do you have any clue how I can fix it? TY.

Sure thing!

**Comment by fraidev** [comment]
> Hi @EstebanBorai. I just noticed there is a problem with the PR. Do you have any clue how I can fix it? TY.

I believe that rebasing master should fix it

File: crates/smartmodule-development-kit/src/clean.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#[derive(Debug, Parser)]
+pub struct CleanCmd {
+    /// Extra arguments to be passed to cargo
+    #[arg(raw = true)]
+    extra_arguments: Vec<String>,
+}
+
+impl CleanCmd {
+    pub(crate) fn process(self) -> Result<()> {
+        println!("{:?}", self.extra_arguments);
```
Comment (EstebanBorai) [inline crates/smartmodule-development-kit/src/clean.rs:18]: I think this can be removed!

**Review by EstebanBorai — COMMENTED** [review]
Nice contribution @diogopontual! Keep it up!

I think it would be nice to have a CLI test as part of this change, that ensures the behavior of `smdk clean` is applied to the project.

CLI tests are developed using BATS in our Fluvio repo, you can find examples here: https://github.com/infinyon/fluvio/blob/master/tests/cli/smdk_smoke_tests/smdk-basic.bats

One way to test `smdk clean` would be to make sure the `target` directory is removed after executing the command.

**Comment by digikata** [comment]
The previous failures were, I think from a github outage, re-running gives an error on fvm, but that is likely because we are about to do a release and this branch needs a rebase.  I'm hitting update on a rebase and we'll see if it's clean.

File: crates/smartmodule-development-kit/src/clean.rs (hunk 1 above)
Comment (diogopontual) [reply crates/smartmodule-development-kit/src/clean.rs]: That horrible println is removed, sorry. I will create the test as soon as I can free up some time. Thank for your help so far.

Comment (EstebanBorai) [reply crates/smartmodule-development-kit/src/clean.rs]: No worries! Thanks so much for taking the time!

Comment (diogopontual) [reply crates/smartmodule-development-kit/src/clean.rs]: Hey @EstebanBorai. I am facing issues running the CLI tests locally. I have no previous experience with bats. Do you have any clue? 
image

**Comment by EstebanBorai** [comment]
Hi @diogopontual!

Seems like we need to run `cargo fmt` on this repo to fix code formatting issues!

File: crates/smartmodule-development-kit/src/clean.rs (hunk 1 above)
Comment (diogopontual) [reply crates/smartmodule-development-kit/src/clean.rs]: NVM. It is working now. I just did 'fluvio cluster start' and the file missing file was created.

**Comment by EstebanBorai** [comment]
Hi @diogopontual! Can you please rebase to have Rust 1.80 updates applied please?

**Review by EstebanBorai — APPROVED** [review]

**Comment by EstebanBorai** [comment]
Great progress @diogopontual! I think we are just missing smoke tests for this feature and we are good to go!

**Comment by EstebanBorai** [comment]
> Wow. is that a green mark? God is good!
😂
 
> I will work on the smoke test this weekend. The week was full. TY.
Great! Ask any questions!

**Comment by diogopontual** [comment]
Hey @EstebanBorai, I did create the test as you described. TY.

**Comment by EstebanBorai** [comment]
> Hey @EstebanBorai, I did create the test as you described. TY.

Thanks to you!

**Review by EstebanBorai — APPROVED** [review]
LGTM! Thanks for your time on this contribution @diogopontual!
Keep it up!

**Comment by EstebanBorai** [comment]
@diogopontual I have updated the PR title to make it match conventional commits!

**Pull request merged**